### PR TITLE
Initial Commit for adding support for web3 providers outside of infur…

### PR DESCRIPTION
**Pull request by raghava-pamula**
Initial Commit for adding support for web3 providers outside of infura. Currently, the project only supports infura web3 providers, and more flexibility is needed since infura has had outages.

https://github.com/Uniswap/routing-api/issues/38

### CODE REVIEW

File: bin/app.ts [hunk 1]
```diff
@@ -93,8 +97,8 @@ export class RoutingAPIPipeline extends Stack {
     // Secrets are stored in secrets manager in the pipeline account. Accounts we deploy to
     // have been granted permissions to access secrets via resource policies.
 
-    const infuraProjectId = sm.Secret.fromSecretAttributes(this, 'InfuraProjectId', {
-      secretCompleteArn: 'arn:aws:secretsmanager:us-east-2:644039819003:secret:infuraProjectId-UlSwK2',
+    const jsonRpcProvider = sm.Secret.fromSecretAttributes(this, 'jsonRpcProvider', {
+      secretCompleteArn: 'arn:aws:secretsmanager:us-east-2:644039819003:secret:jsonRpcProvider-UlSwK2',
```
Comment (willpote) [inline bin/app.ts:101]: Did you create the secret in secrets manager? I don't see it there

File: bin/stacks/routing-lambda-stack.ts [hunk 2]
```diff
@@ -53,6 +58,21 @@ export class RoutingLambdaStack extends cdk.NestedStack {
...
+      POOL_CACHE_BUCKET: poolCacheBucket.bucketName,
+      POOL_CACHE_BUCKET_2: poolCacheBucket2.bucketName,
+      POOL_CACHE_KEY: poolCacheKey,
+      TOKEN_LIST_CACHE_BUCKET: tokenListCacheBucket.bucketName,
+      ETH_GAS_STATION_INFO_URL: ethGasStationInfoUrl,
+    }
+
+    jsonRpcProviderOverride.forEach((url: string, chainId: ChainId) => {
+      ;(env as any)[`JSON_RPC_PROVIDER_${ID_TO_NETWORK_NAME(chainId).toUpperCase}`] = url
+    })
```
Comment (willpote) [inline bin/stacks/routing-lambda-stack.ts:74]: I don't see this used? I think its best to just pass in the ID anyway, then compute the url for each chain

**Review by willpote — CHANGES_REQUESTED** [review]
Make sure you run the integration tests with `npm run integ-tests`. Pretty sure this will fail on all networks except mainnet

File: bin/app.ts (hunk 1 above)
Comment (raghava-pamula) [reply bin/app.ts:101]: https://github.com/Uniswap/routing-api/pull/47/files

Hey reworked and simplified this PR entirely

**Comment by raghava-pamula** [comment]
> Make sure you run the integration tests with `npm run integ-tests`. Pretty sure this will fail on all networks except mainnet

I don't know what URL to use for archival node integ-tests, let's discuss that tomorrow. Here is the reworked PR: https://github.com/Uniswap/routing-api/pull/47/files

**Pull request closed without merging**